src/napari/layers/base/base.py

```diff
@@ -117,6 +116,167 @@ def __call__(self, *args, **kwargs):
         return obj
 
 
+class LayerSlicer(ABC):
+    layer: Layer
+    slice_done = Signal()
+
+    def __init__(self, layer: Layer, data: LayerDataType, cache: bool):
+        self.layer = layer
+        self.dask_optimized_slicing = configure_dask(data, cache)
+        self._slice_input = _SliceInput(
+            ndisplay=2,
+            world_slice=_ThickNDSlice.make_full(ndim=self.ndim),
+            order=tuple(range(self.ndim)),
+        )
+
+    def set_view_slice(self) -> None:
+        with self.dask_optimized_slicing():
+            self._set_view_slice()
+
+    @property
+    def data_slice(self) -> _ThickNDSlice:
+        """Slice in data coordinates."""
+        if len(self._slice_input.not_displayed) == 0:
+            # all dims are displayed dimensions
+            # early return to avoid evaluating data_to_world.inverse
+            return _ThickNDSlice.make_full(point=(np.nan,) * self.ndim)
+
+        return self._slice_input.data_slice(
+            self.layer._data_to_world.inverse,
+        )
+
+    def update_dims(self):
+        self._slice_input = self._slice_input.with_ndim(self.ndim)
+
+    def set_slice_input_from_dims(self, dims: Dims, force: bool) -> bool:
+        slice_input = self.make_slice_input(dims)
+        return self.set_slice_input(slice_input, force)
+
+    def set_slice_input(self, slice_input: _SliceInput, force: bool) -> bool:
+        if self._slice_input == slice_input and not force:
+            return False
+        self._slice_input = copy.deepcopy(slice_input)
+        if self.layer.visible and not force:
+            self.set_view_slice()
+        return True
```

Comment:
`force` probably needs to be propagated to `set_slice_input` like it was in the original code. Also we should add a comment about why the `visible` check is here (cause it wasn't there in the original one).